user-service/src/main/java/rs/edu/raf/banka1/services/implementations/OrderServiceImpl.java

```diff
@@ -1,130 +1,278 @@
     private final MarketService marketService;
+
+    private final TransactionService transactionService;
+    private final CapitalService capitalService;
+
     private final ScheduledExecutorService executorService = Executors.newScheduledThreadPool(1);
     private final Random random = new Random();
+    private final TaskScheduler taskScheduler;
+    private final Double PERCENT = 0.1;
+
+    //Map for storing ScheduledFuture instances for every order - used for task cancellation (kills finished simulation threads).
+    //Used only for orders simulated by this one server instance.
+    private final Map<Long, ScheduledFuture<?>> scheduledFutureMap;
 
     public OrderServiceImpl(
         final OrderMapper orderMapper,
         final OrderRepository orderRepository,
-        final MarketService marketService
+        final MarketService marketService,
+        final TaskScheduler taskScheduler,
+        final TransactionService transactionService,
+        final CapitalService capitalService
     ) {
         this.orderMapper = orderMapper;
         this.orderRepository = orderRepository;
         this.marketService = marketService;
+        this.taskScheduler = taskScheduler;
+        this.transactionService = transactionService;
+        this.capitalService = capitalService;
+
+        scheduledFutureMap = new HashMap<>();
+    }
+
+    @Override
+    public void createOrder(final CreateOrderRequest request, final Employee currentAuth) {
+        final MarketOrder order = orderMapper.requestToMarketOrder(request);
+        ListingBaseDto listingBaseDto;
+
+        if(order.getListingType().equals(ListingType.STOCK)) {
+            listingBaseDto = marketService.getStockById(order.getListingId());
+        } else if(order.getListingType().equals(ListingType.FOREX)) {
+            listingBaseDto = marketService.getForexById(order.getListingId());
+        } else {
+            listingBaseDto = marketService.getFutureById(order.getListingId());
+        }
+
+        if(listingBaseDto == null) return;
+
+        order.setPrice(calculatePrice(order,listingBaseDto,order.getContractSize()));
+        order.setFee(calculateFee(request.getLimitValue(), order.getPrice()));
+        order.setOwner(currentAuth);
+
+        if(!currentAuth.getPosition().equalsIgnoreCase(Constants.SUPERVIZOR)) {
+            if(currentAuth.getOrderlimit() < currentAuth.getLimitNow() + order.getPrice()) {
+                order.setStatus(OrderStatus.DENIED);
+            }
+        }
+
+        if (!orderRequiresApprove(currentAuth)) {
+            order.setStatus(OrderStatus.APPROVED);
+        } else {
+            order.setStatus(OrderStatus.PROCESSING);
+        }
+        orderRepository.save(order);
+
+        //Will automatically throw an exception if there is insufficient capital to create order
+        if(order.getStatus().equals(OrderStatus.APPROVED)) {
+            reserveStockCapital(order);
+        }
+
+        startOrderSimulation(order.getId());
```

Comment:
Da li negde proveravamo koji listing je u pitanju pošto smo rekli da ne radimo simulaciju za `Forex` i `Future`?

Reply:
To se radi u simulaciji. Ako je Forex i Future, automatski ce fulfillovati order skroz

Reply:
Oukee